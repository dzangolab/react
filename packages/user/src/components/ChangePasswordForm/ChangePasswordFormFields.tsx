import { FormActions, Password, useFormContext } from "@dzangolab/react-form";
import { useTranslation } from "@dzangolab/react-i18n";
import React from "react";

interface Properties {
  loading?: boolean;
}

const ChangePasswordFormFields = ({ loading }: Properties) => {
  const { t } = useTranslation("user");

  const {
    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    formState: { errors, submitCount, isDirty },
    register,
    getFieldState,
  } = useFormContext();

  return (
    <>
      <Password
        autoComplete="current-password"
        label={t("changePassword.form.oldPassword.label")}
        name="oldPassword"
        register={register}
        getFieldState={getFieldState}
      />
      <Password
        label={t("changePassword.form.newPassword.label")}
        name="password"
        register={register}
        getFieldState={getFieldState}
      />
      <Password
        label={t("changePassword.form.confirmPassword.label")}
        name="confirmPassword"
        register={register}
        getFieldState={getFieldState}
      />

      <FormActions
        actions={[
          {
            id: "submit",
            label: t("changePassword.form.actions.submit"),
          },
        ]}
        loading={loading}
        alignment="fill"
      />
    </>
  );
};

export default ChangePasswordFormFields;
