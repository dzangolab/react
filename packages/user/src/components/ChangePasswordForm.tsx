import { Provider, FormActionOptions } from "@dzangolab/react-form";
import { useTranslation } from "@dzangolab/react-i18n";
import React from "react";
import * as zod from "zod";

import { ChangePasswordFormData } from "@/views/ChangePassword";

import ChangePasswordFormFields from "./ChangePasswordFormFields";
import { PasswordConfirmationSchema } from "./schemas";

interface Properties {
  handleSubmit: (
    data: ChangePasswordFormData,
    options?: FormActionOptions,
  ) => void;
  loading?: boolean;
}

const ChangePasswordForm = ({ handleSubmit, loading }: Properties) => {
  const { t } = useTranslation("user");

  const ChangePasswordFormSchema = zod
    .object({
      oldPassword: zod
        .string()
        .nonempty(t("changePassword.messages.validation.oldPassword")),
      ...PasswordConfirmationSchema({
        passwordValidationMessage: t(
          "changePassword.messages.validation.mustContain",
        ),
        passwordRequiredMessage: t(
          "changePassword.messages.validation.newPassword",
        ),
        confirmPasswordRequiredMessage: t(
          "changePassword.messages.validation.confirmPassword",
        ),
      }),
    })
    .refine(
      (data) => {
        return data.password === data.confirmPassword;
      },
      {
        message: t("changePassword.messages.validation.mustMatch"),
        path: ["confirmPassword"],
      },
    );

  return (
    <Provider
      validationSchema={ChangePasswordFormSchema}
      onSubmit={(data: ChangePasswordFormData, options?: FormActionOptions) =>
        handleSubmit(
          { oldPassword: data.oldPassword, password: data.password },
          options,
        )
      }
    >
      <ChangePasswordFormFields loading={loading} />
    </Provider>
  );
};

export default ChangePasswordForm;
