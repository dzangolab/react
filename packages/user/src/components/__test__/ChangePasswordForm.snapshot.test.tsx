import { Provider } from "@dzangolab/react-form";
import { render } from "@testing-library/react";
import React from "react";
import { expect, test } from "vitest";

import ChangePasswordForm from "../ChangePasswordForm";

test("Component matches snapshot", () => {
  const { container } = render(
    // eslint-disable-next-line @typescript-eslint/no-empty-function
    <Provider onSubmit={() => {}}>
      <ChangePasswordForm />
    </Provider>,
  );

  expect(container).toMatchSnapshot();
});
