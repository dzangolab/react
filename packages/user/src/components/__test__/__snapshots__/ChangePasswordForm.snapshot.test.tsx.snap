// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Component matches snapshot 1`] = `
<form
  className=""
  noValidate={true}
  onSubmit={[Function]}
>
  <div
    className="field oldPassword"
  >
    <label
      htmlFor="input-field-oldPassword"
    >
      changePassword.form.oldPassword.label
    </label>
    <div
      className="input-field-password"
    >
      <input
        id="input-field-oldPassword"
        name="oldPassword"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder=""
        type="password"
      />
      <span
        className="eye-icon"
        onClick={[Function]}
      >
        <i
          className="pi pi-eye-slash"
        />
      </span>
    </div>
  </div>
  <div
    className="field password"
  >
    <label
      htmlFor="input-field-password"
    >
      changePassword.form.newPassword.label
    </label>
    <div
      className="input-field-password"
    >
      <input
        id="input-field-password"
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder=""
        type="password"
      />
      <span
        className="eye-icon"
        onClick={[Function]}
      >
        <i
          className="pi pi-eye-slash"
        />
      </span>
    </div>
  </div>
  <div
    className="field confirmPassword"
  >
    <label
      htmlFor="input-field-confirmPassword"
    >
      changePassword.form.confirmPassword.label
    </label>
    <div
      className="input-field-password"
    >
      <input
        id="input-field-confirmPassword"
        name="confirmPassword"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder=""
        type="password"
      />
      <span
        className="eye-icon"
        onClick={[Function]}
      >
        <i
          className="pi pi-eye-slash"
        />
      </span>
    </div>
  </div>
  <button
    className="dz-button primary filled medium"
    type="submit"
  >
    <span
      className="label"
    >
      changePassword.form.actions.submit
    </span>
  </button>
</form>
`;
